backends/cuda-gen/ceed-cuda-gen-operator-build.cpp

```diff
@@ -785,15 +785,17 @@ extern "C" int CeedCudaGenOperatorBuild(CeedOperator op) {
   code << "\n#define CeedPragmaSIMD\n";
 
   // Find dim and Q1d
-  bool collograd = false;
+  bool collograd = true;
   for (CeedInt i = 0; i < numinputfields; i++) {
     ierr = CeedOperatorFieldGetBasis(opinputfields[i], &basis); CeedChk(ierr);
     if (basis != CEED_BASIS_COLLOCATED) {
       ierr = CeedBasisGetData(basis, &basis_data); CeedChk(ierr);
+      ierr = CeedQFunctionFieldGetEvalMode(qfinputfields[i], &emode);
+      CeedChk(ierr);
 
       // Check for collocated gradient
-      if (basis_data->d_collograd1d)
-        collograd = true; 
+      if (emode == CEED_EVAL_GRAD)
+        collograd = collograd && !!basis_data->d_collograd1d; 
```

Comment:
Why do you need the double negation here?

Reply:
I put it in there to explicitly cast as a bool. Its a style thing, so we can drop it if preferred.

Reply:
I find it a bit confusing.

Reply:
Do you really to cast if you don't do this?

Reply:
It doesn't change the output at all, so we can drop it

Reply:
Isn't it an `||` instead of `&&`?

Reply:
like this you can use `bool collograd = false`.

Reply:
Previously it was, but I believe that was not fully correct. We can only use the collograd approach if it's supported by all bases applying a grad, I believe

Reply:
`collograd` will be `true` for a mass operator the way it is defined.

Reply:
Same with the current code too. The big question is when do we actually want to use this mode.

Note, while we're here, I should probably generalize this for future nontensor elements. This approach won't work for nontensor and tensor elements both

Reply:
(I'm not sure it's really a bad thing)

Reply:
What `collograd` really means is, do we treat call the qFunction on 1 quadrature point, or on Q1d quadrature points.